### PR TITLE
Wrap the phi of track parameters in the range of [-pi,pi]

### PR DESCRIPTION
**Pull request by beomki-yeo**
This is for constraining the values of phi from Kalman updater and smoother to be within [-pi,pi].
If we don't do this, very occasionally, the while loop of[ `wrap_to_pi` of `is_same_angle`](https://github.com/acts-project/traccc/blob/bc55abf73a82625497a63e3a50d8ddb179a227b4/performance/src/performance/details/is_same_angle.cpp#L16) can take very long time.

### CODE REVIEW

**Review by krasznaa — APPROVED** [review]
It would've been nicer to make this into member code inside of `detray::bound_track_parameters`, but I guess this is the next best thing. 🤔

**Comment by stephenswat** [comment]
Wouldn't it be better to use fmod (
https://en.cppreference.com/w/c/numeric/math/fmod) in the method that has
to wrap the phi values?

**Comment by beomki-yeo** [comment]
I think this PR is using `fmod`

File: core/include/traccc/edm/track_parameters.hpp [hunk 1]
```diff
@@ -31,4 +32,20 @@ using bound_covariance = bound_track_parameters::covariance_type;
 using bound_track_parameters_collection_types =
     collection_types<bound_track_parameters>;
 
+// Wrap the phi of track parameters to [-pi,pi]
+TRACCC_HOST_DEVICE
+inline void wrap_phi(bound_track_parameters& param) {
+
+    traccc::scalar phi = param.phi();
+    static constexpr traccc::scalar TWOPI =
+        2. * traccc::constant<traccc::scalar>::pi;
```
Comment (niermann999) [inline core/include/traccc/edm/track_parameters.hpp:41]:
```suggestion
        2.f * traccc::constant<traccc::scalar>::pi;
```

**Pull request merged**